components/public-api/gitpod/v1/envvar.proto

```diff
@@ -0,0 +1,88 @@
+syntax = "proto3";
+
+package gitpod.v1;
+
+option go_package = "github.com/gitpod-io/gitpod/components/public-api/go/v1";
+
+service EnvironmentVariableService {
+  // ListEnvironmentVariables returns all environment variables for the
+  // authenticated user.
+  //
+  // +return NOT_FOUND User does not have access to a workspace with the given
+  // ID +return NOT_FOUND Workspace does not exist
+  rpc ListEnvironmentVariables(ListEnvironmentVariablesRequest)
+      returns (ListEnvironmentVariablesResponse) {}
+
+  // UpdateEnvironmentVariable updates an environment variable for the
+  // authenticated user.
+  rpc UpdateEnvironmentVariable(UpdateEnvironmentVariableRequest)
+      returns (UpdateEnvironmentVariableResponse) {}
+
+  // CreateEnvironmentVariable creates a new environment variable for the
+  // authenticated user.
+  rpc CreateEnvironmentVariable(CreateEnvironmentVariableRequest)
+      returns (CreateEnvironmentVariableResponse) {}
+
+  // DeleteEnvironmentVariable deletes an environment variable for the
+  // authenticated user.
+  rpc DeleteEnvironmentVariable(DeleteEnvironmentVariableRequest)
+      returns (DeleteEnvironmentVariableResponse) {}
+}
+
+message UserScope {
+  string repository_pattern = 1;
+}
+
+message ConfigurationScope {
+  string configuration_id = 1;
+  bool censored = 2;
```

Comment:
@svenefftinge we could also think to have `EnvironmentVariableAdmission` enum instead, we use it as pattern for visibility in other places.

like
```proto
enum EnvironmentVariableAdmission {
  
  ENVIRONMENT_VARIABLE_ADMISSION_UNSPECIFIED;
  // only prebuild
  ENVIRONMENT_VARIABLE_ADMISSION_PREBUILD;
  // prebuild + workspace
  ENVIRONMENT_VARIABLE_ADMISSION_EVERYWHERE; // or   ENVIRONMENT_VARIABLE_ADMISSION_WORKSPACE;
}
```

Reply:
Yes, that makes sense. we might have more scenarios in future

Reply:
How are these enum values to current `censored` value?
true === ENVIRONMENT_VARIABLE_ADMISSION_PREBUILD and false === ENVIRONMENT_VARIABLE_ADMISSION_EVERYWHERE?

Reply:
censored is ENVIRONMENT_VARIABLE_ADMISSION_PREBUILD